crates/forge_loader/src/manifest.rs

```diff
@@ -208,63 +282,102 @@ impl<T> AsRef<T> for FunctionTy<T> {
-            .chain(
-                self.event_triggers
-                    .into_iter()
-                    .map(|trigger| trigger.raw.function),
-            )
-            .chain(
-                self.workflow_post_functions
-                    .into_iter()
-                    .map(|pf| pf.raw.function),
-            )
-            .collect();
-
-        let mut alternate_functions = vec![];
-        for module in self.extra.into_values().flatten() {
-            alternate_functions.extend(module.function);
-            if let Some(resolver) = module.resolver {
-                alternate_functions.push(resolver.function);
-            }
-        }
+        // Get all the Triggers and represent them as a new struct thing where "webtrigger" attribute is true
+        // for all trigger things
+        // Get all the Triggers and represent them as a new struct thing where "webtrigger" attribute is true
+        // for all trigger things
 
-        self.consumers.iter().for_each(|consumer| {
-            if !alternate_functions.contains(&consumer.resolver.function) {
-                ignored_functions.insert(consumer.resolver.function);
-            }
+        let mut invokable_functions = BTreeSet::new();
+
+        self.data_provider.iter().for_each(|dataprovider| {
+            invokable_functions.extend(dataprovider.callback);
         });
 
-        self.functions.into_iter().filter_map(move |func| {
-            if ignored_functions.contains(&func.key) {
-                return None;
-            }
-            Some(
-                if self
-                    .webtriggers
-                    .binary_search_by_key(&func.key, |trigger| trigger.function)
-                    .is_ok()
-                {
-                    FunctionTy::WebTrigger(func)
-                } else {
-                    FunctionTy::Invokable(func)
-                },
-            )
+        self.custom_field.iter().for_each(|customfield| {
+            invokable_functions.extend(customfield.value);
+            invokable_functions.extend(customfield.search_suggestions);
+            invokable_functions.extend(customfield.edit);
+
+            invokable_functions.insert(customfield.common_keys.function);
+            invokable_functions.extend(customfield.common_keys.resolver);
+        });
+
+        self.ui_modifications.iter().for_each(|ui| {
+            invokable_functions.insert(ui.common_keys.function);
+            invokable_functions.extend(ui.common_keys.resolver);
+        });
+
+        self.workflow_validator.iter().for_each(|validator| {
+            invokable_functions.insert(validator.common_keys.key);
+
+            invokable_functions.insert(validator.common_keys.function);
+
+            invokable_functions.extend(validator.common_keys.resolver);
+        });
+
+        self.workflow_post_function
+            .iter()
+            .for_each(|post_function| {
+                invokable_functions.insert(post_function.common_keys.key);
```

Comment:
<!--
 Copyright 2024 Joshua Wong.
 SPDX-License-Identifier: Apache-2.0
-->
The `key` property isn't a function, it's the identifier for the module, so it shouldn't be in the set of invokable functions.